pandas/core/indexes/base.py

```diff
@@ -360,7 +375,7 @@ def __new__(cls, data=None, dtype=None, copy=False, name=None,
                 if all(isinstance(e, tuple) for e in data):
                     from .multi import MultiIndex
                     return MultiIndex.from_tuples(
-                        data, names=name or kwargs.get('names'))
+                        data, names=names or kwargs.get('names') or name)
```

Comment:
``kwargs.get('names')`` shouldn't be needed any more, right? Moreover, I understand ``name`` was accepted as a substitute for ``names`` (as in ``name=['level1', 'level2']``)... but once we correctly support ``names`` thanks to this PR, we shouldn't any more allow this, I think, and this could just be ``names=names``.

(You are right that instead testing the length of names to be 1 is really messy)

Reply:
@toobaz, thanks a lot for a note, you're right :)
One more question then: should we remove it from `MultiIndex` too?

https://github.com/pandas-dev/pandas/blob/78c3ff97a8795c2de9861b6b0a40a5fa4b8e331f/pandas/core/indexes/multi.py#L139-L140